cmdutil/debug/debug.go

```diff
@@ -71,3 +76,88 @@ func (s *Server) Stop(_ error) {
 
 	close(s.done)
 }
+
+// PProfServer wraps a pprof server.
+type PProfServer struct {
+	logger        logrus.FieldLogger
+	addr          string
+	done          chan struct{}
+	pprofServer   *http.Server
+	profileConfig PProfServerConfig
+}
+
+// ProfileConfig holds the configuration for the pprof server.
+type PProfServerConfig struct {
+	Addr                 string
+	ProfileNames         []string
+	MutexProfileFraction int
+}
+
+// NewPProfServer sets up a pprof server with configurable profiling types and returns a PProfServer instance.
+func NewPProfServer(config PProfServerConfig, l logrus.FieldLogger) *PProfServer {
+	if config.Addr == "" {
+		config.Addr = "127.0.0.1:9998" // Default port
+	}
+
+	// Create a new HTTP mux for handling pprof routes.
+	mux := http.NewServeMux()
+
+	// Iterate over the handlers and add them to the mux.
+	for _, profile := range config.ProfileNames {
+		if profile == "mutex" {
+			if config.MutexProfileFraction == 0 {
+				config.MutexProfileFraction = 2 // Use default value of 2 if not set
```

Comment:
Let's not modify the config. That is unexpected behavior. Instead, we can use a local variable:
```
mpf := defaultMutextProfileFraction // value is 2
if config.MutexProfileFraction != 0 {
  mpf = config.MutextProfileFraction
}
runtime.SetMutextProfileFraction(mpf)
```